File: Software/MenuPrototype/Controls/TextInfo.cs
using System;

namespace MenuPrototype
{
	partial class Program
	{
		////////////////////////////////////////////////////////////////////
		public class TextInfo : MenuItem
		{
			String _prompt;
			Func<string> _displayFunc;

			public TextInfo(String prompt, Func<string> displayFunc) : base("")
			{
				_prompt = prompt;
				_displayFunc = displayFunc;
			}

			public override void onDisplay(bool modal = false)
			{
				Console.WriteLine(_prompt + _displayFunc() + "            ");
			}
		}
	}
}

File: Software/Arduino code/OpenAstroTracker/lib/menu/controls/TextInfo.cpp
#include "MainMenu.hpp"
#include "TextInfo.hpp"

TextInfo::TextInfo(String prompt, displayFunc displayFunc) : MenuItem("", "")
{
    _prompt = prompt;
    _displayFunc = displayFunc;
}

void TextInfo::onDisplay(bool modal)
{
    getMainMenu()->writeToLCD(0, 1, _prompt + _displayFunc());
}
